File: leetcode/1217_play_with_chips.py
# -*- coding: utf-8 -*-

from __future__ import absolute_import, division, print_function


class Solution(object):
    def minCostToMoveChips(self, chips):
        count_even, count_odd = 0, 0
        for chip in chips:
            if chip % 2 == 0:
                count_even += 1
            else:
                count_odd += 1

        return min(count_even, count_odd)


if __name__ == '__main__':
    solution = Solution()

    assert 1 == solution.minCostToMoveChips([1, 2, 3])
    assert 2 == solution.minCostToMoveChips([2, 2, 2, 3, 3])

File: leetcode/1217_play_with_chips.cpp
// Copyright (c) 2019 Jacopo Notarstefano

#include <algorithm>
#include <cassert>
#include <vector>

using namespace std;

class Solution {
 public:
    int minCostToMoveChips(vector<int>& chips) {
        int count_even = 0, count_odd = 0;
        for (const auto& chip : chips) {
            if (chip % 2 == 0) {
                count_even++;
            } else {
                count_odd++;
            }
        }

        return min(count_even, count_odd);
    }
};

int main() {
    auto solution = Solution();

    vector<int> chips = {1, 2, 3};

    assert(1 == solution.minCostToMoveChips(chips));
}
